File: src/Utils/StackSorter.java
package Utils;

import java.util.Stack;

public class StackSorter {

    public void sortStack(Stack<Integer> originalStack) {
        Stack<Integer> auxStack = new Stack<>();

        while (!originalStack.isEmpty()) {
            
            int temp = originalStack.pop();

            while (!auxStack.isEmpty() && auxStack.peek() > temp) {
                originalStack.push(auxStack.pop());
            }

            auxStack.push(temp);
        }

        while (!auxStack.isEmpty()) {
            originalStack.push(auxStack.pop());
        }
    }
}

File: python/stack_sorter.py
class StackSorter:

    def sort_stack(self, original_stack: list):
       
        aux_stack = []

        while original_stack:
            
            temp = original_stack.pop()

            while aux_stack and aux_stack[-1] > temp:
                original_stack.append(aux_stack.pop())
            
            aux_stack.append(temp)
        
        while aux_stack:
            original_stack.append(aux_stack.pop())